release/internal/hashrelease/pinnedversion.go

```diff
@@ -203,6 +203,7 @@ func RetrievePinnedOperator(outputDir string) (OperatorComponent, error) {
 }
 
 // RetrievePinnedOperatorVersion retrieves the operator version from the pinned version file.
+// TODO: We should get this from metadata.yaml instead of pinned_versions.yaml.
```

Comment:
Yeah, probably not. I'll remove this TODO.

I'd like "pinned_versions.yaml" to go away though. It's a bit of a hack to tie all of this together, but we should be able to determine the operator version and product version in step 1 and pass it down to the places that need it, rather than treating `pinned_versions.yaml` as the source of truth.